### ISSUE TITLE
[PRE REVIEW]: Production Function Estimation in R: The prodest Package

### ISSUE THREAD

**Issue by whedon** [openjournals/joss-reviews#316]
**Submitting author:** @GabrieleRovigatti (<a href="http://orcid.org/0000-0003-4021-3070">Gabriele Rovigatti</a>)
**Repository:** <a href="https://github.com/GabrieleRovigatti/prodest/tree/master/prodest" target ="_blank">https://github.com/GabrieleRovigatti/prodest/tree/master/prodest</a>
**Version:** v0.1.1
**Editor:** @leeper
**Reviewer:** @sebastianbarfort

**Author instructions**

Thanks for submitting your paper to JOSS @GabrieleRovigatti. The JOSS editor (shown at the top of this issue) will work with you on this issue to find a reviewer for your submission before creating the main review issue.

@GabrieleRovigatti if you have any suggestions for potential reviewers then please mention them here in this thread. In addition, [this list of people](https://github.com/openjournals/joss/blob/master/docs/reviewers.csv) have already agreed to review for JOSS and may be suitable for this submission.

**Editor instructions**

The JOSS submission bot @whedon is here to help you find and assign reviewers and start the main review. To find out what @whedon can do for you type:

```
@whedon commands
```

**Comment by whedon** [comment]
Hello human, I'm @whedon. I'm here to help you with some common editorial tasks for JOSS.

For a list of things I can do to help you, just type:

```
@whedon commands
```

**Comment by whedon** [comment]
Here are some things you can ask me to do:

```
# List all of Whedon's capabilities
@whedon commands

# Assign a GitHub user as the reviewer of this submission
@whedon assign @username as reviewer

# List the GitHub usernames of the JOSS editors
@whedon list editors

# List of JOSS reviewers together with programming language preferences and domain expertise
@whedon list reviewers

# Change editorial assignment
@whedon assign @username as editor

# Set the software archive DOI at the top of the issue e.g.
@whedon set 10.0000/zenodo.00000 as archive

# Open the review issue
@whedon start review
```

:construction: **Important** :construction:

This is all quite new. Please make sure you check the top of the issue after running a @whedon command (you might also need to refresh the page to see the issue update).

**Comment by GabrieleRovigatti** [comment]
@whedon assign @robohyun66 as reviewer

**Comment by whedon** [comment]
I'm sorry @GabrieleRovigatti, I'm afraid I can't do that. That's something only JOSS editors are allowed to do.

**Comment by GabrieleRovigatti** [comment]
@whedon list editors

**Comment by whedon** [comment]
Current JOSS editors:

```
@acabunoc
@arfon
@arokem
@biorelated
@brainstorm
@cMadan
@danielskatz
@jakevdp
@karthik
@katyhuff
@Kevin-Mattheus-Moerman
@kyleniemeyer
@labarba
@mgymrek
@pjotrp
@tracykteal
```

**Comment by GabrieleRovigatti** [comment]
@whedon start review

**Comment by whedon** [comment]
I'm sorry @GabrieleRovigatti, I'm afraid I can't do that. That's something only JOSS editors are allowed to do.

**Comment by arfon** [comment]
> @whedon start review

@GabrieleRovigatti - we'll need one of the JOSS editors to volunteer to edit this submission before we can proceed.

**Comment by GabrieleRovigatti** [comment]
@afron just trying how @whedon commands work. I'll wait for someone to accept the task.

Thanks

**Comment by GabrieleRovigatti** [comment]
@afron sounds like no-one is volunteering. I can't see any editor with the necessary economic background. Wrong journal maybe? Any suggestion?

**Comment by arfon** [comment]
@GabrieleRovigatti - yes, I'm sorry this is taking a while. We don't really have anyone with economics background on the editorial board currently. If you're willing to wait I'm happy to go and try and search out someone.

**Comment by GabrieleRovigatti** [comment]
@arfon I am willing to wait as long as it is necessary! I am sorry if this is going to take long / will need a supplementary load of work for you. I like your journal, its 'mission' and main goals, but maybe I should have read more carefully the editorial board background before the submission. Hopefully you will find someone soon - tip: most people working in the relevant literature use Stata for panel-data analyses like the one I propose, maybe you may try among Stata-users? 

By the way, sorry for mispelling your name (twice!) previously, I just figured out the exact spelling with your last email.

**Comment by arfon** [comment]
> @arfon I am willing to wait as long as it is necessary! I am sorry if this is going to take long / will need a supplementary load of work for you. I like your journal, its 'mission' and main goals, but maybe I should have read more carefully the editorial board background before the submission.

Thanks for your patience. We're shortly going to be adding a new editor to help with submissions like yours :-)

> By the way, sorry for mispelling your name (twice!) previously, I just figured out the exact spelling with your last email.

No problem. It's a common mistake!

**Comment by arfon** [comment]
@leeper - would you be able to edit this submission for JOSS?

**Comment by leeper** [comment]
@whedon assign @leeper as editor

**Comment by whedon** [comment]
OK, the editor is @leeper

**Comment by leeper** [comment]
@GabrieleRovigatti I'm not seeing a `paper.md` file in the repository. Can you add one per [the author guidelines](http://joss.theoj.org/about#author_guidelines)? An example paper is [available here](https://github.com/arfon/fidgit/tree/master/paper) if you'd like to see what we have in mind.

**Comment by arfon** [comment]
> @GabrieleRovigatti I'm not seeing a paper.md file in the repository. Can you add one per the author guidelines? An example paper is available here if you'd like to see what we have in mind.

Thanks @leeper. Also, the section [_'what should my paper contain'_](http://joss.theoj.org/about#paper_structure) here has some guidance for the sort of content we're looking to be present in the `paper.md`.

Many thanks!

**Comment by GabrieleRovigatti** [comment]
@leeper sorry for that, I was sure to have already uploaded the paper and the .bib file. Hopefully, now you can find everything [here](https://github.com/GabrieleRovigatti/prodest/tree/master/prodest/paper) 

@arfon I tried to follow the guidelines and - at the same time - to be as concise as possible, hoping that interested readers will refer to the vignette included in the package itself.

**Comment by leeper** [comment]
@GabrieleRovigatti Great. Thanks.

**Comment by leeper** [comment]
@whedon assign @sebastianbarfort as reviewer

**Comment by whedon** [comment]
OK, the reviewer is @sebastianbarfort

**Comment by leeper** [comment]
Thanks for your submission @GabrieleRovigatti. I've confirmed that this submission has:

- [X]  An OSI approved license (GPL)
- [X] A description of the software and its research applications
- [X] Is in scope for this journal

Two things:

 1. I might be inclined to add just a sentence or two of additional text to the paper.md to contextualize the software for a general audience that may not know what Total Factor Productivity (TFP) estimation is.
 2. Are you submitting both the R and Stata versions as one package or just the R version (as linked above)?

@sebastianbarfort has agreed to review.

**Comment by GabrieleRovigatti** [comment]
@leeper thanks for having started the review - and for accepting the task in the first place! And thanks @sebastianbarfort, too, for his time and effort. 

1. I added a sentence explaining - in a word - what is meant by TFP. 

2. I am submitting the R routine only. I already submitted the Stata version to the Stata Journal. By the way, I am not sure Stata routines comply with the 'open source' nature of the journal, or would you accept proprietary software packages, too?

**Comment by leeper** [comment]
@GabrieleRovigatti Okay, great. Let's proceed on R only.

Side note: @arfon @karthik we have a good question on journal scope. If the software is open source but it's an add-on to a proprietary tool (like Stata) is that in or out of scope for JOSS?

**Comment by leeper** [comment]
@whedon start review magic-word=bananas

**Comment by whedon** [comment]
OK, I've started the review over in https://github.com/openjournals/joss-reviews/issues/371. Feel free to close this issue now!

**Comment by leeper** [comment]
@sebastianbarfort Review checklist is up at: https://github.com/openjournals/joss-reviews/issues/371

**Comment by karthik** [comment]
@leeper 


We did have a very long thread about this (I'll forward it to you). But the outcome was:

> We strongly prefer software that doesn't rely upon proprietary environments.
Open source submissions that do rely upon a proprietary environment/tool are acceptable provided that it is possible to find a reviewer for the submission and that the submitting author is responsible for finding these reviewers.
Add to the reviewer guidelines:
That the reviewer should encourage the author to consider porting their tools to be compatible with open source/free variants of the runtime e.g. Octave instead of MATLAB. (This won't be a condition of acceptance however).

**Comment by leeper** [comment]
@karthik Thanks. I'll close this then since the review issue is now open and all seems well here.
